src/test/lib/ApiTest.cpp

```diff
@@ -1847,6 +1847,86 @@ void QuicTestConfigurationSetParam()
     }
 }
 
+// Used by Listener and Connection
+void CibirIDTests(HQUIC Handle, uint32_t Param) {
+    //
+    // buffer length test
+    //
+    {
+        TestScopeLogger LogScope("Buffer length test");
+        //
+        // Buffer is bigger than QUIC_MAX_CIBIR_LENGTH + 1
+        //
+        {
+            TestScopeLogger LogScope("Buffer is bigger than QUIC_MAX_CIBIR_LENGTH + 1");
+            uint8_t Cibir[128] = {0};
+            TEST_QUIC_STATUS(
+                QUIC_STATUS_INVALID_PARAMETER,
+                MsQuic->SetParam(
+                    Handle,
+                    Param,
+                    sizeof(Cibir),
+                    &Cibir));
+        }
+
+        //
+        // BufferLength == 1
+        //
+        {
+            TestScopeLogger LogScope("BufferLength == 1");
+            uint8_t Cibir[1] = {0};
+            TEST_QUIC_STATUS(
+                QUIC_STATUS_INVALID_PARAMETER,
+                MsQuic->SetParam(
+                    Handle,
+                    Param,
+                    sizeof(Cibir),
+                    &Cibir));
+        }
+
+        //
+        // Good without value, length 0
+        //
+        {
+            TestScopeLogger LogScope("no value, Bufferlength == 0");
+            TEST_QUIC_SUCCEEDED(
+                MsQuic->SetParam(
+                    Handle,
+                    Param,
+                    0,
+                    nullptr));
+        }
+    }
+
+    //
+    // Buffer starts from non-zero is not supported
```

Comment:
I've had this thought a couple of times on other tests/PRs, but what do we want to do if/when we actually start supporting some of these negative test cases? Eventually we would like to support non-zero offsets for instance.

Reply:
What I could do is to add comment for the time when someone try to enable the feature to remove the test without concern.